### PR TITLE
sink/codec(ticdc): remove useless MixedBuild and Reset funcs

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/4423

### What is changed and how it works?

remove useless MixedBuild and Reset funcs. We have removed the cdc log, so this is completely useless code.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has interface methods change

Side effects

None

Related changes

None

### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/sink/codec/json.go [hunk 1]
```diff
@@ -540,70 +507,14 @@ func (d *JSONEventBatchEncoder) EncodeDDLEvent(e *model.DDLEvent) (*MQMessage, e
...
 // Size implements the EventBatchEncoder interface
 func (d *JSONEventBatchEncoder) Size() int {
-	return d.keyBuf.Len() + d.valueBuf.Len()
-}
-
-// Reset implements the EventBatchEncoder interface
-func (d *JSONEventBatchEncoder) Reset() {
-	d.keyBuf.Reset()
-	d.valueBuf.Reset()
+	return -1
```
Comment (amyangfei) [inline cdc/sink/codec/json.go:517]: Is there any side effect to `return -1` here (before it is removed)?

Comment (hi-rustin) [reply cdc/sink/codec/json.go:517]: No, because the size of our original json protocol is only for mix encoder. So there is no problem to return -1.

**Review by amyangfei — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Pull request merged**